src/main/java/com/fluxninja/aperture/armeria/DecoratingHTTPClient.java

```diff
@@ -0,0 +1,69 @@
+package com.fluxninja.aperture.armeria;
+
+import com.fluxninja.aperture.sdk.ApertureSDK;
+import com.fluxninja.aperture.sdk.ApertureSDKException;
+import com.fluxninja.aperture.sdk.Flow;
+import com.fluxninja.aperture.sdk.FlowStatus;
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.client.SimpleDecoratingHttpClient;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DecoratingHTTPClient extends SimpleDecoratingHttpClient {
+    private ApertureSDK apertureSDK;
+    private String featureName;
+
+    public DecoratingHTTPClient(HttpClient delegate, ApertureSDK apertureSDK, String featureName) {
+        super(delegate);
+        try {
+            this.apertureSDK = apertureSDK;
+            this.featureName = featureName;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Exception {
+        Map<String, String> labels = labelsFromRequest(req);
+        Flow flow = this.apertureSDK.startFlow(featureName, labels);
+
+        if (flow.accepted()) {
+            HttpResponse res;
+            try {
+                res = unwrap().execute(ctx, req);
+                flow.end(FlowStatus.OK);
+            } catch (InterruptedException | ApertureSDKException e) {
+                e.printStackTrace();
+                return HttpResponse.of(HttpStatus.BAD_REQUEST);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+            return res;
+        } else {
+            try {
+                flow.end(FlowStatus.Error);
+            } catch (ApertureSDKException e) {
+                e.printStackTrace();
+            }
+            return HttpResponse.of(HttpStatus.BAD_REQUEST);
```

Comment:
Need to set response code based on [CheckResponse.RejectReason](https://github.com/fluxninja/aperture/blob/main/api/aperture/flowcontrol/v1/flowcontrol.proto#L33)

429 Too many requests in case of `REJECT_REASON_RATE_LIMITED`
503 Service unavailable in case of `REJECT_REASON_CONCURRENCY_LIMITED`

Reply:
Should we also use http codes for RPC responses?